### PR TITLE
[infra-proxy-service] updates orgs test cases & validations

### PR DESCRIPTION
**Pull request by vsingh-msys**
### :nut_and_bolt: Description: What code changed, and why?

- Used validation helpers for orgs params validations.
- Minor refactoring orgs test cases, might we can match exact string but skipping here and added similar assert check like `servers_test.go`.

### :chains: Related Resources
Closes: https://github.com/chef/automate/issues/4423

### :+1: Definition of Done
- better & reused orgs validations.

### :athletic_shoe: How to Build and Test the Change
- `make test`

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
Signed-off-by: Vivek Singh <vivek.singh@msystechnologies.com>

### CODE REVIEW

File: components/infra-proxy-service/server/orgs_test.go [hunk 1]
```diff
@@ -532,11 +532,22 @@ func TestOrgs(t *testing.T) {
...
+			grpctest.AssertCode(t, codes.InvalidArgument, err)
+		})
+
 		t.Run("when the org does not exists, return org not found", func(t *testing.T) {
 			ctx := context.Background()
 			orgID := "97e01ea1-976e-4626-88c8-43345c5d934f"
 			resp, err := cl.GetOrg(ctx, &request.GetOrg{
-				Id: orgID,
+				Id:       orgID,
+				ServerId: serverRes.Server.Id,
```
Comment (vsingh-msys) [inline components/infra-proxy-service/server/orgs_test.go:550]: Added missing validation for server ID while fetching the org.

File: components/infra-proxy-service/server/orgs_test.go [hunk 2]
```diff
@@ -1031,7 +1042,7 @@ func TestOrgs(t *testing.T) {
 				Projects:  []string{},
 			})
 			require.Nil(t, resp)
-			assert.Contains(t, err.Error(), "must supply org ID")
+			assert.Error(t, err, "must supply org ID")
```
Comment (vsingh-msys) [inline components/infra-proxy-service/server/orgs_test.go:1045]: Loosely coupled assert to check the Err object only, avoiding hardcoded error message check here, any thoughts would be helpful!

File: components/infra-proxy-service/server/server.go [hunk 3]
```diff
@@ -16,7 +16,7 @@ import (
 	"github.com/chef/automate/lib/tracing"
 )
 
-// NewGRPCServer creates a grpc server that serves all Teams GRPC APIs
+// NewGRPCServer creates a grpc server that serves all infra-proxy-service GRPC APIs
```
Comment (vsingh-msys) [inline components/infra-proxy-service/server/server.go:19]: minor typo

**Review by stevendanna — APPROVED** [review]
I missed the PR for the new validation library, nice work there. Glad to see us spreading its use.

File: components/infra-proxy-service/server/orgs.go [hunk 4]
```diff
@@ -3,45 +3,29 @@ package server
...
-	if req.AdminKey == "" {
-		s.service.Logger.Debug("incomplete create org request: missing org admin key")
-		return nil, status.Error(codes.InvalidArgument, "must supply org admin key")
-	}
+	// Validate all request fields are required
+	err := validation.New(validation.Options{
+		Target:          "org",
+		Request:         *req,
+		RequiredDefault: true,
+	}).Validate()
```
Comment (msorens) [inline components/infra-proxy-service/server/orgs.go:25]: Looking over the validation library internals a bit, which I had not seen before. Looks like `RequiredDefault` says that every field of the `Request` must be non-empty, is that correct?
Then I am not clear on the second loop inside `Validate()` which seems to be doing the same check on `Rules` (though I do not see you using `Rules` anywhere in the codebase). 
Ideally would be good if you could add a couple comment lines inside validation.go::Validate that addresses my questions.

Comment (vsingh-msys) [reply components/infra-proxy-service/server/orgs.go:25]: Yes, this validation library added to provide the validation out of the box. If `RequiredDefault` is set the all request field must be non-empty.


The second loop is the use case if the user wants to validate only specific fields to non-empty or other validations(TODO).

Name would be non-empty
```

err = validation.New(validation.Options{
		Target:          "org",
		Request:         *req,
		Rules: validation.Rules{
			"Name": []string{"required"}, // ip_address, email or multiple rules we can be add here but is in future scope.
		},
	}).Validate()
```


Will created further PRs and provides the doc & a couple of optimization for its uses. Thanks!

**Review by msorens — APPROVED** [review]
Looks good.

**Pull request merged**